examples/fabric-admin/device_manager/DeviceSynchronization.cpp

```diff
@@ -128,35 +128,35 @@ void DeviceSynchronizer::OnAttributeData(const ConcreteDataAttributePath & path,
 void DeviceSynchronizer::OnReportEnd()
 {
     // Report end is at the end of all attributes (success)
+    MoveToState(State::ReceivedResponse);
+}
+
+void DeviceSynchronizer::OnDone(chip::app::ReadClient * apReadClient)
+{
 #if defined(PW_RPC_ENABLED)
-    if (!DeviceMgr().IsCurrentBridgeDevice(mCurrentDeviceData.node_id))
+    if (mState == State::ReceivedResponse && !DeviceMgr().IsCurrentBridgeDevice(mCurrentDeviceData.node_id))
```

Comment:
would it be possible to somehow reduce the deep nesting in here?
Could be done as a followup - not high priority as logic seems reasonable.

Reply:
I will add in follow up PR https://github.com/project-chip/connectedhomeip/issues/35382